packages/openbox.rb

```diff
@@ -0,0 +1,56 @@
+require 'package'
+
+class Openbox < Package                 # The first character of the class name must be upper case
+  description 'Openbox is a highly configurable, next generation window manager with extensive standards support.'
+  homepage 'http://openbox.org/'
+  version '3.6.1'
+  compatibility 'all'
+  source_url 'http://openbox.org/dist/openbox/openbox-3.6.1.tar.xz'
+  source_sha256 'abe75855cc5616554ffd47134ad15291fe37ebbebf1a80b69cbde9d670f0e26d'   # Use the command "sha256sum"
+
+  depends_on 'libev'
+  depends_on 'startup_notification'
+  depends_on 'xcb_util_cursor'
+  depends_on 'xcb_util_keysyms'
+  depends_on 'xcb_util_wm'
+  depends_on 'xcb_util_xrm'
+  depends_on 'yajl'
+  depends_on 'wayland' => :build
+  depends_on 'sommelier'
+  depends_on 'xorg_proto'
+  depends_on 'glib'
+  depends_on 'libxfixes'
+  depends_on 'libsm'
+  depends_on 'pango' => :build
+  depends_on 'libx11' => :build
+
+  ENV['CFLAGS'] = "-lX11 -lXau"
+  def self.build                   # the steps required to build the package
+    system "./configure #{CREW_OPTIONS}"
+    system "make"
+  end
+
+  def self.install                 # the steps required to install the package
+    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+    system "echo '#!/bin/bash' > startopenbox"
+    system "echo 'stopsommelier' >> startopenbox"
+    system "echo 'echo am start x.org.server/.MainActivity | sudo android-sh' >> startopenbox"
+    system "echo 'echo Starting XServer XSDL' >> startopenbox"
+    system "echo 'sleep 10' >> startopenbox"
+    system "echo 'export DISPLAY=100.115.92.2:0' >> startopenbox"
+    system "echo 'openbox' >> startopenbox"
+    system "echo 'echo XServer closed or logged out session' >> startopenbox"
+    system "echo 'initsommelier' >> startopenbox"
```

Comment:
`startsommelier` instead of `initsommelier`?

Reply:
We also need `export DISPLAY=:0`.

Reply:
The reasons of using `startsommelier` is mentioned in Addtional information section, do you know where is the `startsommelier` command path?

Reply:
Just run `crew files sommelier` and it should appear as one of the binaries.

Reply:
> Just run `crew files sommelier` and it should appear as one of the binaries.

Sorry for late reply
Still can't found `startsommelier` with `crew files sommelier` , but I noticed a line in `sommelier` post-install
```shell
echo 'alias startsommelier="set -a && source ~/.sommelier.env && set +a && initsommelier"' >> ~/.bashrc
```
Seems the `startsommelier` command was provide by `alais`

Reply:
startsommelier  sources ~/.sommerlier.env and note that it sets ```export DISPLAY=:0``` there. But this openbox recipe is predicated upon the use of an Android Xserver, hence the ```"echo 'export DISPLAY=100.115.92.2:0' >> startopenbox"``` - so ```startsommelier``` should re replicated in a way that avoids setting the display to ```:0``` from the sourcing of ~/.bashrc.

Reply:
> But this openbox recipe is predicated upon the use of an Android Xserver

We need to re-run `sommelier` after the connect of Xserver connection lost or `openbox` exit. Can I just add `alias startsommelier="set -a && source ~/.sommelier.env && set +a && initsommelier"` to the script directly to set up the `startsommelier`command?

Reply:
Keep in mind that you have startsommelier in your .bashrc by default... and that will run ```set -a && source ~/.sommelier.env && set +a && initsommelier``` .  It might be easier to suggest editing the .bashrc to have startsommelier mean ```set -a && source ~/.sommelier-openbox.env && set +a && initsommelier``` and have your customized DISPLAY setting in ~/.sommelier-openbox.env  .